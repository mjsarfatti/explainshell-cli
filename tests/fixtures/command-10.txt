git log --graph --abbrev-commit --pretty=oneline origin..mybranch