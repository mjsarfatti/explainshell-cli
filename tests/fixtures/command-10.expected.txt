git log

    Show commit logs

    [ git log -> https://explainshell.com/explain/1/git-log ]

--graph

    --graph
        Draw a text-based graphical representation of the commit history on the left hand side of the output.
        This may cause extra lines to be printed in between commits, in order for the graph history to be
        drawn properly.
    
        This enables parent rewriting, see History Simplification below.
    
        This implies the --topo-order option by default, but the --date-order option may also be specified.

--abbrev-commit

    --abbrev-commit
        Instead of showing the full 40-byte hexadecimal commit object name, show only a partial prefix. Non
        default number of digits can be specified with "--abbrev=<n>" (which also modifies diff output, if it
        is displayed).
    
        This should make "--pretty=oneline" a whole lot more readable for people using 80-column terminals.

--pretty=oneline

    Commit Formatting
        --pretty[=<format>], --format=<format>
            Pretty-print the contents of the commit logs in a given format, where <format> can be one of oneline,
            short, medium, full, fuller, email, raw and format:<string>. See the "PRETTY FORMATS" section for
            some additional details for each format. When omitted, the format defaults to medium.
    
            Note: you can specify the default pretty format in the repository configuration (see git-config(1)).

origin..mybranch

    <since>..<until>
        Show only commits between the named two commits. When either <since> or <until> is omitted, it
        defaults to HEAD, i.e. the tip of the current branch. For a more complete list of ways to spell
        <since> and <until>, see gitrevisions(7).
    
    [--] <path>...
        Show only commits that are enough to explain how the files that match the specified paths came to be.
        See "History Simplification" below for details and other simplification modes.
    
        To prevent confusion with options and branch names, paths may need to be prefixed with "-- " to
        separate them from options or refnames. 